packages/mermaid/src/rendering-util/handle-markdown-text.spec.ts

```diff
@@ -215,13 +214,13 @@ test('markdownToLines - No auto wrapping', () => {
     [
       [
         {
-          "content": "Hello, how do",
+          "content": "Hello,&nbsp;how&nbsp;do",
           "type": "normal",
         },
       ],
       [
         {
-          "content": "you do?",
+          "content": "you&nbsp;do?",
```

Comment:
This is a special case where we _don't_ want actual spaces to be present. 
<img width="514" alt="image" src="https://github.com/user-attachments/assets/e05e83bb-7a1d-4157-9c6a-aa53a31a6a32">

We are injecting the `&nbsp;` ourselves here, to avoid line breaks. (which was not happening previously, as the regex wouldn't match)

I think it was not present in the test before, as the spaces were encoded as [`U+00A0`](https://www.compart.com/en/unicode/U+00A0), which is unicode version of `&nbsp;`. This is evident from the VSCode screenshot below. 

<img width="354" alt="image" src="https://github.com/user-attachments/assets/21b8e612-b08d-4f23-9764-c208ce6f7758">

As this is in the internals and not visible to users, `&nbsp;` feels like a better option than unicode.

Reply:
<img width="586" alt="image" src="https://github.com/user-attachments/assets/acb8d320-a7bd-4819-8c7a-a357642ba883">

`&nbsp;` was already being used in other places in the same file.